### ISSUE TITLE
BUG : Grafana RAW dashboards create malformed grafana-rawDashboardDefinitions file

### ISSUE THREAD

**Issue by prune998** [brancz/kubernetes-grafana#79]
Since latest commit [567be6b15b7f3b747c48dc7b111c1860cab121c7](https://github.com/brancz/kubernetes-grafana/commit/567be6b15b7f3b747c48dc7b111c1860cab121c7) a file `grafana-rawDashboardDefinitions` is created with the content :
```
[ ]

```
when `rawDashboards` is empty or not specified.
This is a bad yaml/json content.

# Reproducing : 

```
mkdir test
cd test
jb init
jb install github.com/coreos/kube-prometheus/jsonnet/kube-prometheus@master
mkdir manifests
cp ../cluster.jsonnet .
jsonnet  -J vendor -m manifests cluster.jsonnet
```
content of `cluster.jsonnet` : 
```
local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';

local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') + (import 'etcd-mixin/mixin.libsonnet') + {
  prometheus+:: {
  },
  _config+:: {
    namespace: 'monitoring',
    },
    grafana+:: {
      config+:{
      // rawDashboards: {},
      datasources: [{
        name: 'prometheus',
        type: 'prometheus',
        access: 'proxy',
        orgId: 1,
        url: 'http://prometheus-k8s.monitoring.svc:9090',
        version: 1,
        isDefault: true,
        editable: true,
      }, ],
      plugins: [
        'grafana-piechart-panel', 'camptocamp-prometheus-alertmanager-datasource'
      ],
    },
  },
};

{ ['00namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
{ ['0prometheus-operator-' + name]: kp.prometheusOperator[name] for name in std.objectFields(kp.prometheusOperator) } +
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
```

# attended result
The file `manifests/grafana-rawDashboardDefinitions` should not be created when `rawDashboards` is empty or not defined.

# solution : 
I wasn't able to find a clean solution where the `manifests/grafana-rawDashboardDefinitions` file is not created when empty. 

Note that there may be the same problem with `manifests/grafana-DashboardDefinitions`, it just never happen as there are some defaults dashboards.

# versions
`Jsonnet commandline interpreter v0.14.0` from `go get` and official `C++` version throuth OsX  `Brew`

**Comment by brancz** [comment]
My bad, we should only create the key in the first place if the rawDashboards are non empty. Do you think you could fix this?

**Comment by prune998** [comment]
Sounds my jsonnet skills are not good enought... I tried with no luck... Anybody else ?

**Comment by johannwillgrubs** [comment]
I'am working on a fix

**Comment by mattayes** [comment]
This was fixed with #80, right?

**Comment by brancz** [comment]
Correct
